### ISSUE TITLE
Graduate Seccomp to GA

### ISSUE THREAD

**Issue by pjbgf** [kubernetes/kubernetes#91286]
Graduate seccomp to GA, as defined on the respective [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190717-seccomp-ga.md). This is an umbrella issue to track all outstanding work.

### Core Items
- [x] API Server changes https://github.com/kubernetes/kubernetes/pull/91408 (MERGED)
- [x] Kubelet changes https://github.com/kubernetes/kubernetes/pull/91381 (MERGED)
- [x] Mark flag `--seccomp-profile-root` as deprecated https://github.com/kubernetes/kubernetes/pull/91182 (MERGED)
- [x] Update PSP to support seccomp fields https://github.com/kubernetes/kubernetes/pull/92856 (MERGED)
- [x] Document Kubernetes seccomp support https://github.com/kubernetes/website/pull/21278 (MERGED)

### Nice to Have on 1.19
- [x] Add seccomp least privilege for docker sandbox https://github.com/kubernetes/kubernetes/pull/90948 (MERGED)
- [x] Add seccomp least privilege for kuberuntime https://github.com/kubernetes/kubernetes/pull/90949 (MERGED)
- [x] E2E tests review https://github.com/kubernetes/kubernetes/pull/91442 (LGTM)

### Pushed to 1.22
- [x] Add Warning mechanism for deprecated features usage (based off [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1693-warnings))

/area kubelet
/area kubelet-api
/sig node
/area security
/assign

**Comment by pjbgf** [comment]
@palnabarun just a heads up that we have concluded all the core items for seccomp to go GA. There are still 2 pending PRs that ideally would also go in 1.19, but we are waiting for them to be reviewed: https://github.com/kubernetes/kubernetes/pull/90948 and https://github.com/kubernetes/kubernetes/pull/90949.

**Comment by kikisdeliveryservice** [comment]
https://github.com/kubernetes/kubernetes/pull/92856#issuecomment-655950700 needs to be rebased apparently and isn't just waiting in the merge pool as verify is not passing.

**Comment by saschagrunert** [comment]
@pjbgf I think the warning mechanism was blocked by a not existing API. This is now tracked in https://github.com/kubernetes/kubernetes/issues/94626

**Comment by pjbgf** [comment]
@saschagrunert  Updated the description to `v1.22` given that the warning mechanism is now set to that milestone.

**Comment by tallclair** [comment]
/lifecycle frozen

What is the remaining work here?

**Comment by saschagrunert** [comment]
@tallclair I think we have to follow-up on https://github.com/kubernetes/kubernetes/pull/109819 and implement: 

> Auto-population of the seccomp fields from the annotations is planned to be removed in 1.27.

I can take care of that.
